### PR TITLE
#621 Crop varieties sidebar

### PR DESCRIPTION
**Pull request by oshiho3**
Please review.

Implemented #621 with javascript as I didn't realize rails-y implementation was preferred over javascript here and I wanted the toggle to look smooth instead of refreshing the whole page.
Let me know if you want me to rewrite using different technology etc.

Thanks!

### CODE REVIEW

File: spec/features/crops/crop_detail_page_spec.rb [hunk 1]
```diff
@@ -4,6 +4,61 @@
...
+    let!(:roma2) { FactoryGirl.create(:crop, :name => 'Roma tomato 2', :parent => crop) }
+    let!(:roma3) { FactoryGirl.create(:crop, :name => 'Roma tomato 3', :parent => crop) }
+    let!(:roma4) { FactoryGirl.create(:crop, :name => 'Roma tomato 4', :parent => crop) }
+
+    scenario "The crop has 4 varieties" do
+
+      visit crop_path(crop)
+
+      # It lists all 5 items (note: including the top level item.)
+      # It DOES NOT have "Show all/less" toggle link
```
Comment (tygriffin) [inline spec/features/crops/crop_detail_page_spec.rb]: You might use a `within` block to focus on that .crop-hierarchy class.

File: spec/features/crops/crop_detail_page_spec.rb [hunk 2]
```diff
@@ -4,6 +4,61 @@
...
+      expect(page).to have_css('li', text: /tomato/i, count: 5)
+      expect(page).to have_no_css('button', text: /Show all+/i)
+      expect(page).to have_no_css('button', text: /Show less+/i)
```
Comment (tygriffin) [inline spec/features/crops/crop_detail_page_spec.rb]: Does it make more sense to use `have_selector` and `have_no_selector` here?

File: spec/features/crops/crop_detail_page_spec.rb (hunk 1 above)
Comment (oshiho3) [inline spec/features/crops/crop_detail_page_spec.rb]: Good suggestion, thanks! Done now.

File: spec/features/crops/crop_detail_page_spec.rb (hunk 2 above)
Comment (oshiho3) [inline spec/features/crops/crop_detail_page_spec.rb]: Thanks, done. I guess I was originally trying to match its classes as well so I thought I needed have_css, but decided to omit it. So you're right have_selector does the job.

**Comment by oshiho3** [comment]
Code has been updated to incorporate @tygriffin's review and also
- Added a couple of more test scenarios
- Made crop/varieties partial to accept <code>max</code> param which specifies how many items can be seen without "Show all" button
- Number of crops to include the number of descendants below child as well

**Comment by pozorvlak** [comment]
Looks good to me - moving into release 9.

**Pull request merged**